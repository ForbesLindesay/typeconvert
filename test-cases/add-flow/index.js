import addA from './add-a';
import addB from './add-b';

export {addA, addB};