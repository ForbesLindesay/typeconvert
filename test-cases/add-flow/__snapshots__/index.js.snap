// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`./add-a.js.flow ast 1`] = `
Object {
  "kind": "File",
  "leadingComments": Array [],
  "loc": SourceLocation {
    "end": Position {
      "column": 0,
      "line": 11,
    },
    "start": Position {
      "column": 0,
      "line": 1,
    },
  },
  "statements": Array [
    Object {
      "body": undefined,
      "id": Object {
        "kind": "Identifier",
        "leadingComments": Array [],
        "loc": SourceLocation {
          "end": Position {
            "column": 50,
            "line": 9,
          },
          "identifierName": "add",
          "start": Position {
            "column": 17,
            "line": 9,
          },
        },
        "name": "add",
      },
      "kind": "FunctionDeclaration",
      "leadingComments": Array [
        Object {
          "kind": "CommentLine",
          "loc": SourceLocation {
            "end": Position {
              "column": 8,
              "line": 1,
            },
            "start": Position {
              "column": 0,
              "line": 1,
            },
          },
          "value": " @flow",
        },
        Object {
          "kind": "CommentBlock",
          "loc": SourceLocation {
            "end": Position {
              "column": 3,
              "line": 8,
            },
            "start": Position {
              "column": 0,
              "line": 3,
            },
          },
          "value": "*
 * Adds two numbers together
 *
 * @param a The first number
 * @param b The second number
 ",
        },
      ],
      "loc": SourceLocation {
        "end": Position {
          "column": 51,
          "line": 9,
        },
        "start": Position {
          "column": 0,
          "line": 9,
        },
      },
      "params": Array [
        Object {
          "kind": "FunctionParameter",
          "leadingComments": Array [],
          "loc": SourceLocation {
            "end": Position {
              "column": 30,
              "line": 9,
            },
            "start": Position {
              "column": 21,
              "line": 9,
            },
          },
          "name": Object {
            "kind": "Identifier",
            "leadingComments": Array [],
            "loc": SourceLocation {
              "end": Position {
                "column": 22,
                "line": 9,
              },
              "identifierName": "a",
              "start": Position {
                "column": 21,
                "line": 9,
              },
            },
            "name": "a",
          },
          "optional": false,
          "type": Object {
            "kind": "StringTypeAnnotation",
            "leadingComments": Array [],
            "loc": SourceLocation {
              "end": Position {
                "column": 30,
                "line": 9,
              },
              "start": Position {
                "column": 24,
                "line": 9,
              },
            },
          },
        },
        Object {
          "kind": "FunctionParameter",
          "leadingComments": Array [],
          "loc": SourceLocation {
            "end": Position {
              "column": 41,
              "line": 9,
            },
            "start": Position {
              "column": 32,
              "line": 9,
            },
          },
          "name": Object {
            "kind": "Identifier",
            "leadingComments": Array [],
            "loc": SourceLocation {
              "end": Position {
                "column": 33,
                "line": 9,
              },
              "identifierName": "b",
              "start": Position {
                "column": 32,
                "line": 9,
              },
            },
            "name": "b",
          },
          "optional": false,
          "type": Object {
            "kind": "StringTypeAnnotation",
            "leadingComments": Array [],
            "loc": SourceLocation {
              "end": Position {
                "column": 41,
                "line": 9,
              },
              "start": Position {
                "column": 35,
                "line": 9,
              },
            },
          },
        },
      ],
      "restParam": undefined,
      "returnType": Object {
        "kind": "StringTypeAnnotation",
        "leadingComments": Array [],
        "loc": SourceLocation {
          "end": Position {
            "column": 50,
            "line": 9,
          },
          "start": Position {
            "column": 44,
            "line": 9,
          },
        },
      },
      "typeParams": Array [],
    },
    Object {
      "expression": Object {
        "kind": "Identifier",
        "leadingComments": Array [],
        "loc": SourceLocation {
          "end": Position {
            "column": 18,
            "line": 10,
          },
          "identifierName": "add",
          "start": Position {
            "column": 15,
            "line": 10,
          },
        },
        "name": "add",
      },
      "kind": "ExportDefault",
      "leadingComments": Array [],
      "loc": SourceLocation {
        "end": Position {
          "column": 19,
          "line": 10,
        },
        "start": Position {
          "column": 0,
          "line": 10,
        },
      },
    },
  ],
}
`;

exports[`./add-b.js ast 1`] = `
Object {
  "kind": "File",
  "leadingComments": Array [],
  "loc": SourceLocation {
    "end": Position {
      "column": 0,
      "line": 12,
    },
    "start": Position {
      "column": 0,
      "line": 1,
    },
  },
  "statements": Array [
    Object {
      "body": Array [
        Object {
          "argument": Object {
            "kind": "BinaryExpression",
            "leadingComments": Array [],
            "left": Object {
              "kind": "Identifier",
              "leadingComments": Array [],
              "loc": SourceLocation {
                "end": Position {
                  "column": 10,
                  "line": 10,
                },
                "identifierName": "a",
                "start": Position {
                  "column": 9,
                  "line": 10,
                },
              },
              "name": "a",
            },
            "loc": SourceLocation {
              "end": Position {
                "column": 14,
                "line": 10,
              },
              "start": Position {
                "column": 9,
                "line": 10,
              },
            },
            "operator": "+",
            "right": Object {
              "kind": "Identifier",
              "leadingComments": Array [],
              "loc": SourceLocation {
                "end": Position {
                  "column": 14,
                  "line": 10,
                },
                "identifierName": "b",
                "start": Position {
                  "column": 13,
                  "line": 10,
                },
              },
              "name": "b",
            },
          },
          "kind": "ReturnStatement",
          "leadingComments": Array [],
          "loc": SourceLocation {
            "end": Position {
              "column": 15,
              "line": 10,
            },
            "start": Position {
              "column": 2,
              "line": 10,
            },
          },
        },
      ],
      "id": Object {
        "kind": "Identifier",
        "leadingComments": Array [],
        "loc": SourceLocation {
          "end": Position {
            "column": 27,
            "line": 9,
          },
          "identifierName": "add",
          "start": Position {
            "column": 24,
            "line": 9,
          },
        },
        "name": "add",
      },
      "kind": "FunctionDeclaration",
      "leadingComments": Array [
        Object {
          "kind": "CommentLine",
          "loc": SourceLocation {
            "end": Position {
              "column": 8,
              "line": 1,
            },
            "start": Position {
              "column": 0,
              "line": 1,
            },
          },
          "value": " @flow",
        },
        Object {
          "kind": "CommentBlock",
          "loc": SourceLocation {
            "end": Position {
              "column": 3,
              "line": 8,
            },
            "start": Position {
              "column": 0,
              "line": 3,
            },
          },
          "value": "*
 * Adds two numbers together
 *
 * @param a The first number
 * @param b The second number
 ",
        },
      ],
      "loc": SourceLocation {
        "end": Position {
          "column": 1,
          "line": 11,
        },
        "start": Position {
          "column": 15,
          "line": 9,
        },
      },
      "params": Array [
        Object {
          "kind": "FunctionParameter",
          "leadingComments": Array [],
          "loc": SourceLocation {
            "end": Position {
              "column": 37,
              "line": 9,
            },
            "identifierName": "a",
            "start": Position {
              "column": 28,
              "line": 9,
            },
          },
          "name": Object {
            "kind": "Identifier",
            "leadingComments": Array [],
            "loc": SourceLocation {
              "end": Position {
                "column": 37,
                "line": 9,
              },
              "identifierName": "a",
              "start": Position {
                "column": 28,
                "line": 9,
              },
            },
            "name": "a",
          },
          "optional": false,
          "type": Object {
            "kind": "StringTypeAnnotation",
            "leadingComments": Array [],
            "loc": SourceLocation {
              "end": Position {
                "column": 37,
                "line": 9,
              },
              "start": Position {
                "column": 31,
                "line": 9,
              },
            },
          },
        },
        Object {
          "kind": "FunctionParameter",
          "leadingComments": Array [],
          "loc": SourceLocation {
            "end": Position {
              "column": 48,
              "line": 9,
            },
            "identifierName": "b",
            "start": Position {
              "column": 39,
              "line": 9,
            },
          },
          "name": Object {
            "kind": "Identifier",
            "leadingComments": Array [],
            "loc": SourceLocation {
              "end": Position {
                "column": 48,
                "line": 9,
              },
              "identifierName": "b",
              "start": Position {
                "column": 39,
                "line": 9,
              },
            },
            "name": "b",
          },
          "optional": false,
          "type": Object {
            "kind": "StringTypeAnnotation",
            "leadingComments": Array [],
            "loc": SourceLocation {
              "end": Position {
                "column": 48,
                "line": 9,
              },
              "start": Position {
                "column": 42,
                "line": 9,
              },
            },
          },
        },
      ],
      "restParam": undefined,
      "returnType": Object {
        "kind": "StringTypeAnnotation",
        "leadingComments": Array [],
        "loc": SourceLocation {
          "end": Position {
            "column": 57,
            "line": 9,
          },
          "start": Position {
            "column": 51,
            "line": 9,
          },
        },
      },
      "typeParams": Array [],
    },
    Object {
      "expression": Object {
        "kind": "Identifier",
        "leadingComments": Array [],
        "loc": SourceLocation {
          "end": Position {
            "column": 27,
            "line": 9,
          },
          "identifierName": "add",
          "start": Position {
            "column": 24,
            "line": 9,
          },
        },
        "name": "add",
      },
      "kind": "ExportDefault",
      "leadingComments": Array [],
      "loc": SourceLocation {
        "end": Position {
          "column": 1,
          "line": 11,
        },
        "start": Position {
          "column": 0,
          "line": 9,
        },
      },
    },
  ],
}
`;

exports[`./index.js ast 1`] = `
Object {
  "kind": "File",
  "leadingComments": Array [],
  "loc": SourceLocation {
    "end": Position {
      "column": 20,
      "line": 4,
    },
    "start": Position {
      "column": 0,
      "line": 1,
    },
  },
  "statements": Array [
    Object {
      "id": Object {
        "kind": "Identifier",
        "leadingComments": Array [],
        "loc": SourceLocation {
          "end": Position {
            "column": 11,
            "line": 1,
          },
          "identifierName": "addA",
          "start": Position {
            "column": 7,
            "line": 1,
          },
        },
        "name": "addA",
      },
      "kind": "ImportStatement",
      "leadingComments": Array [],
      "loc": SourceLocation {
        "end": Position {
          "column": 27,
          "line": 1,
        },
        "start": Position {
          "column": 0,
          "line": 1,
        },
      },
      "mode": "value",
      "source": Object {
        "id": Object {
          "kind": "Identifier",
          "leadingComments": Array [],
          "loc": SourceLocation {
            "end": Position {
              "column": 11,
              "line": 1,
            },
            "start": Position {
              "column": 7,
              "line": 1,
            },
          },
          "name": "default",
        },
        "kind": "EsImportSource",
        "leadingComments": Array [],
        "loc": SourceLocation {
          "end": Position {
            "column": 11,
            "line": 1,
          },
          "start": Position {
            "column": 7,
            "line": 1,
          },
        },
        "relativePath": "./add-a",
      },
    },
    Object {
      "id": Object {
        "kind": "Identifier",
        "leadingComments": Array [],
        "loc": SourceLocation {
          "end": Position {
            "column": 11,
            "line": 2,
          },
          "identifierName": "addB",
          "start": Position {
            "column": 7,
            "line": 2,
          },
        },
        "name": "addB",
      },
      "kind": "ImportStatement",
      "leadingComments": Array [],
      "loc": SourceLocation {
        "end": Position {
          "column": 27,
          "line": 2,
        },
        "start": Position {
          "column": 0,
          "line": 2,
        },
      },
      "mode": "value",
      "source": Object {
        "id": Object {
          "kind": "Identifier",
          "leadingComments": Array [],
          "loc": SourceLocation {
            "end": Position {
              "column": 11,
              "line": 2,
            },
            "start": Position {
              "column": 7,
              "line": 2,
            },
          },
          "name": "default",
        },
        "kind": "EsImportSource",
        "leadingComments": Array [],
        "loc": SourceLocation {
          "end": Position {
            "column": 11,
            "line": 2,
          },
          "start": Position {
            "column": 7,
            "line": 2,
          },
        },
        "relativePath": "./add-b",
      },
    },
    Object {
      "exportedName": Object {
        "kind": "Identifier",
        "leadingComments": Array [],
        "loc": SourceLocation {
          "end": Position {
            "column": 12,
            "line": 4,
          },
          "identifierName": "addA",
          "start": Position {
            "column": 8,
            "line": 4,
          },
        },
        "name": "addA",
      },
      "kind": "ExportNamed",
      "leadingComments": Array [],
      "loc": SourceLocation {
        "end": Position {
          "column": 12,
          "line": 4,
        },
        "start": Position {
          "column": 8,
          "line": 4,
        },
      },
      "localName": Object {
        "kind": "Identifier",
        "leadingComments": Array [],
        "loc": SourceLocation {
          "end": Position {
            "column": 12,
            "line": 4,
          },
          "identifierName": "addA",
          "start": Position {
            "column": 8,
            "line": 4,
          },
        },
        "name": "addA",
      },
    },
    Object {
      "exportedName": Object {
        "kind": "Identifier",
        "leadingComments": Array [],
        "loc": SourceLocation {
          "end": Position {
            "column": 18,
            "line": 4,
          },
          "identifierName": "addB",
          "start": Position {
            "column": 14,
            "line": 4,
          },
        },
        "name": "addB",
      },
      "kind": "ExportNamed",
      "leadingComments": Array [],
      "loc": SourceLocation {
        "end": Position {
          "column": 18,
          "line": 4,
        },
        "start": Position {
          "column": 14,
          "line": 4,
        },
      },
      "localName": Object {
        "kind": "Identifier",
        "leadingComments": Array [],
        "loc": SourceLocation {
          "end": Position {
            "column": 18,
            "line": 4,
          },
          "identifierName": "addB",
          "start": Position {
            "column": 14,
            "line": 4,
          },
        },
        "name": "addB",
      },
    },
  ],
}
`;
